### DISCUSSION TITLE
objdetect.jpg - Writing to an events Notes during an event or after with a delay?

### DISCUSSION THREAD

**Discussion by baudneo** [ZoneMinder/zoneminder#3951] [General]
I am trying to understand the timing of when I should write to an events Notes. This is for the objdetect.jpg popup.

What I think is happening is, when the ML runs quickly and is complete before an event itself is done, the notes that the ML writes are overwritten by the event closing. I am just wondering if that's when the event writes its notes to DB? I could probably use EventStopCommand to be a trigger to write notes, but it may need a delay to make sure that the `:detected:` keyword is in the event Notes.

@connortechnology

**Comment by connortechnology** [comment]
While an event is recording, it can and will update the Notes column. It has been on my todo list for a while to address this.  The Event Tags feature goes some way, and I also added Event_Data.  The idea being to NOT update the main event record, but to just insert new data.  Helps ease contention on the Events table too.  So yeah, at this time you cannot really alter the Notes column until the event is finished.  How about we move away from using the Notes field entirely.

**Reply by baudneo** [reply]
I agree with moving away, I was just hooking into existing logic for objdetect.jpg.

**Reply by baudneo** [reply]
How do you feel about the possibility of configuring ML via the web gui? I still haven't tackled learning JS/html/htmx so idk if now is the best time. Just thought I would ask if that's something we can work on in the future?

**Reply by connortechnology** [reply]
We definitely can.  I'm trying to hire some people and @IgorA100 and @Simpler1 are active here on the front end.  I want to move on all the things we talked about before.  I have no time, but am starting to have money.  I at least want to get a 4000 series nvidia card going here and start training a model on the local cats and birds.  We can start by adding an AI tab to monitor edit, with a toggle for doing ai.  Then we need inputs for which models to run... which objects to recognize, etc.  Gets to be a pretty complex UI.

**Reply by baudneo** [reply]
KK, i've had the ML on the back burner since before xmas. Its stable, fast and works but it isnt where I want it to be yet. Ive got time to sink into things again.

**Reply by connortechnology** [reply]
Please come back to discord or slack or whatever so our chat is a little easier and less out in the open.

**Reply by IgorA100** [reply]
I haven’t tested object detection yet and I don’t even understand how it works :)
But to the best of my ability, I hope I can provide all possible assistance.
But I have problems with having free time.

**Reply by baudneo** [reply]
I think im still in slack, havent used it in awhile. WIll be on today after work!

@IgorA100 - the goal is to make the ML easy as possible to get running and configure. That way you dont need to know the intricacies, just enjoy having it work for you. Glad you're here to help!
